### PR TITLE
Applies changes of @jusdon to dev

### PR DESCRIPTION
**Pull request by timo2o1o**
Changes from 21.02.2024

DONE:
- Implemented Azure Microsoft Identity authentication flow.
- Created an order list table with pagination proof of concept.
- Handled order HTTP requests to list all orders.
- Updated API requests on the order index to implement pagination.
- Do some frontend stuff on order index page

@timo2o1o, could you please review the IAccService / AccService? Is the way I implemented API pagination for orders satisfactory for you? If yes, I would like to do the same for incoming and outgoing invoices.

### CODE REVIEW

File: ACCDataAdapter/DPApp/SQLCommands/InitializeMasterdataTables.sql
Comment (timo2o1o) [inline ACCDataAdapter/DPApp/SQLCommands/InitializeMasterdataTables.sql:1]: This is the wrong place for this file.

File: ACCDataAdapter/Properties/Resources.resx [hunk 1]
```diff
@@ -128,7 +128,7 @@
     <value>..\DPApp\SQLCommands\ReadAusgangsrechnung.sql;System.String, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089;Windows-1252</value>
   </data>
   <data name="ReadAusgangsrechnungen" type="System.Resources.ResXFileRef, System.Windows.Forms">
-    <value>..\DPApp\SQLCommands\ReadAusgangsrechnungen.sql;System.String, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089;utf-8</value>
+    <value>..\DPApp\SQLCommands\ReadAusgangsrechnungen.sql;System.String, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089;Windows-1252</value>
```
Comment (timo2o1o) [inline ACCDataAdapter/Properties/Resources.resx:131]: These changes need to be reverted

File: ACCWebAPI/Controllers/IncomingInvoiceController.cs [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        private readonly IACCService _accService;
+
+        public IncomingInvoiceController(ILogger<IncomingInvoiceController> logger, IACCService accService)
+        {
+            _logger = logger;
+            _accService = accService;
+        }
+
+        [HttpGet("")]
+        public async Task<IActionResult> Index()
```
Comment (timo2o1o) [inline ACCWebAPI/Controllers/IncomingInvoiceController.cs:23]: Lets rename this method to meet the .Net naming guidelines: GetIncomingInvoices()

File: ACCWebAPI/Controllers/IncomingInvoiceController.cs [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+            }
+            catch (Exception ex)
+            {
+                return StatusCode(500, ex.Message);
+            }
+        }
+
+        [HttpPost("")]
+        public async Task<IActionResult> Store(Eingangsrechnung eingangsrechnung)
```
Comment (timo2o1o) [inline ACCWebAPI/Controllers/IncomingInvoiceController.cs:39]: Lets also rename this method to meet the .Net naming guidelines: PostInvoiceToDB()

File: ACCWebAPI/Controllers/IncomingInvoiceController.cs [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+                return Ok(neueEingangsrechnung);
+            }
+            catch (Exception ex)
+            {
+                return StatusCode(500, ex.Message);
+            }
+        }
+
+        [HttpGet("{IncomingInvoiceID}")]
+        public async Task<IActionResult> Show(int IncomingInvoiceID)
```
Comment (timo2o1o) [inline ACCWebAPI/Controllers/IncomingInvoiceController.cs:58]: I think you get it... Lets name this GetIncomingInvoice

File: ACCWebAPI/Controllers/IncomingInvoiceController.cs [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+                return Ok(eingangsrechnung);
+            }
+            catch (Exception ex)
+            {
+                return StatusCode(500, ex.Message);
+            }
+        }
+
+        [HttpPut("{IncomingInvoiceID}")]
+        public async Task<IActionResult> Update(int IncomingInvoiceID, Eingangsrechnung eingangsrechnung)
```
Comment (timo2o1o) [inline ACCWebAPI/Controllers/IncomingInvoiceController.cs:72]: ... PutIncomingInvoice()

File: ACCWebAPI/Controllers/IncomingInvoiceController.cs [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+                return Ok(eingangsrechnungUpdated);
+            }
+            catch (Exception ex)
+            {
+                return StatusCode(500, ex.Message);
+            }
+        }
+
+        [HttpDelete("{IncomingInvoiceID}")]
+        public async Task<IActionResult> Destroy(int IncomingInvoiceID)
```
Comment (timo2o1o) [inline ACCWebAPI/Controllers/IncomingInvoiceController.cs:89]: DeleteIncomingInvoice()

File: ACCWebAPI/Controllers/OrderController.cs [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+        private readonly IACCService _accService;
+
+        public OrderController(ILogger<OrderController> logger, IACCService accService)
+        {
+            _logger = logger;
+            _accService = accService;
+        }
+
+        [HttpGet("")]
+        public async Task<IActionResult> Index(int? currentPage, int? pageSize)
```
Comment (timo2o1o) [inline ACCWebAPI/Controllers/OrderController.cs:24]: ... GetOrders()

File: ACCWebAPI/Controllers/OrderController.cs [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+                return Ok(response);
+            }
+            catch (Exception ex)
+            {
+                return StatusCode(500, ex.Message);
+            }
+        }
+
+        [HttpPost("")]
+        public async Task<IActionResult> Store(Auftrag auftrag)
```
Comment (timo2o1o) [inline ACCWebAPI/Controllers/OrderController.cs:46]: PostOrder()

File: ACCWebAPI/Controllers/OrderController.cs [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+                return Ok(neuerAuftrag);
+            }
+            catch (Exception ex)
+            {
+                return StatusCode(500, ex.Message);
+            }
+        }
+
+        [HttpGet("{OrderID}")]
+        public async Task<IActionResult> Show(int OrderID)
```
Comment (timo2o1o) [inline ACCWebAPI/Controllers/OrderController.cs:65]: GetOrder()

File: ACCWebAPI/Controllers/OrderController.cs [hunk 10]
```diff
@@ -0,0 +1,113 @@
...
+                return Ok(auftrag);
+            }
+            catch (Exception ex)
+            {
+                return StatusCode(500, ex.Message);
+            }
+        }
+
+        [HttpPut("{OrderID}")]
+        public async Task<IActionResult> Update(int OrderID, Auftrag auftrag)
```
Comment (timo2o1o) [inline ACCWebAPI/Controllers/OrderController.cs:79]: PutOrder()

File: ACCWebAPI/Controllers/OrderController.cs [hunk 11]
```diff
@@ -0,0 +1,113 @@
...
+                return Ok(auftragUpdated);
+            }
+            catch (Exception ex)
+            {
+                return StatusCode(500, ex.Message);
+            }
+        }
+
+        [HttpDelete("{OrderID}")]
+        public async Task<IActionResult> Destroy(int OrderID)
```
Comment (timo2o1o) [inline ACCWebAPI/Controllers/OrderController.cs:96]: DeleteOrder()

File: ACCWebAPI/Controllers/OutgoingInvoiceController.cs [hunk 12]
```diff
@@ -0,0 +1,106 @@
...
+        private readonly IACCService _accService;
+
+        public OutgoingInvoiceController(ILogger<OutgoingInvoiceController> logger, IACCService accService)
+        {
+            _logger = logger;
+            _accService = accService;
+        }
+
+        [HttpGet("")]
+        public async Task<IActionResult> Index()
```
Comment (timo2o1o) [inline ACCWebAPI/Controllers/OutgoingInvoiceController.cs:23]: GetInvoices()

File: ACCWebAPI/Controllers/OutgoingInvoiceController.cs [hunk 13]
```diff
@@ -0,0 +1,106 @@
...
+
+            }
+            catch (Exception ex)
+            {
+                return StatusCode(500, ex.Message);
+            }
+        }
+
+        [HttpPost("")]
+        public async Task<IActionResult> Store(Ausgangsrechnung ausgangsrechnung)
```
Comment (timo2o1o) [inline ACCWebAPI/Controllers/OutgoingInvoiceController.cs:39]: PostInvoice()

File: ACCWebAPI/Controllers/OutgoingInvoiceController.cs [hunk 14]
```diff
@@ -0,0 +1,106 @@
...
+                return Ok(neueAusgangsrechnung);
+            }
+            catch (Exception ex)
+            {
+                return StatusCode(500, ex.Message);
+            }
+        }
+
+        [HttpGet("{OutgoingInvoiceID}")]
+        public async Task<IActionResult> Show(int OutgoingInvoiceID)
```
Comment (timo2o1o) [inline ACCWebAPI/Controllers/OutgoingInvoiceController.cs:58]: GetInvoice()

File: ACCWebAPI/Controllers/OutgoingInvoiceController.cs [hunk 15]
```diff
@@ -0,0 +1,106 @@
...
+                return Ok(ausgangsrechnung);
+            }
+            catch (Exception ex)
+            {
+                return StatusCode(500, ex.Message);
+            }
+        }
+
+        [HttpPut("{OutgoingInvoiceID}")]
+        public async Task<IActionResult> Update(int OutgoingInvoiceID, Ausgangsrechnung ausgangsrechnung)
```
Comment (timo2o1o) [inline ACCWebAPI/Controllers/OutgoingInvoiceController.cs:72]: PutInvoice()

File: ACCWebAPI/Controllers/OutgoingInvoiceController.cs [hunk 16]
```diff
@@ -0,0 +1,106 @@
...
+                return Ok(ausgangsrechnungUpdated);
+            }
+            catch (Exception ex)
+            {
+                return StatusCode(500, ex.Message);
+            }
+        }
+
+        [HttpDelete("{OutgoingInvoiceID}")]
+        public async Task<IActionResult> Destroy(int OutgoingInvoiceID)
```
Comment (timo2o1o) [inline ACCWebAPI/Controllers/OutgoingInvoiceController.cs:89]: DeleteInvoice()

File: ACCWebAPI/appsettings.json [hunk 17]
```diff
@@ -1,9 +1,9 @@
 {
   "AzureAd": {
     "Instance": "https://login.microsoftonline.com/",
-    "Domain": "",
-    "TenantId": "",
-    "ClientId": "",
+    "Domain": "btelligent.com",
```
Comment (timo2o1o) [inline ACCWebAPI/appsettings.json:4]: Lets move this stuff to an appsettings.development.json file

File: ACCWebUI/Services/OrderService.cs [hunk 18]
```diff
@@ -0,0 +1,87 @@
...
+
+            return authResult.AccessToken;
+        }
+
+        public async Task<OrderResponseDTO> GetOrdersAsync(int currentPage, int pageSize)
+        {
+
+            _httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", await AcquireToken());
+
+            HttpResponseMessage response = await _httpClient.GetAsync($"http://localhost:32771/orders?currentPage={currentPage}&pageSize={pageSize}");
```
Comment (timo2o1o) [inline ACCWebUI/Services/OrderService.cs:35]: Hard coding service url and port and nailing it to http is not a good idea ;)

**Review by timo2o1o — COMMENTED** [review]
Well done! Just a few things to change before we can merge

File: ACCDataAdapter/DPApp/SQLCommands/InitializeMasterdataTables.sql
Comment (timo2o1o) [reply ACCDataAdapter/DPApp/SQLCommands/InitializeMasterdataTables.sql:1]: I deleted it.

File: ACCDataAdapter/Properties/Resources.resx (hunk 1 above)
Comment (timo2o1o) [reply ACCDataAdapter/Properties/Resources.resx]: Reverted it.

File: ACCWebAPI/Controllers/AccountingUnitController.cs [hunk 19]
```diff
@@ -0,0 +1,114 @@
...
+        private readonly IACCService _accService;
+
+        public AccountingUnitController(ILogger<AccountingUnitController> logger, IACCService accService)
+        {
+            _logger = logger;
+            _accService = accService;
+        }
+
+        [HttpGet("")]
+        public async Task<IActionResult> Index(int? currentPage, int? pageSize)
```
Comment (timo2o1o) [inline ACCWebAPI/Controllers/AccountingUnitController.cs:24]: Could you please stop implementing this methods using these names xD?

File: ACCWebAPI/Controllers/IncomingInvoiceController.cs [hunk 20]
```diff
@@ -0,0 +1,114 @@
...
+        private readonly IACCService _accService;
+
+        public IncomingInvoiceController(ILogger<IncomingInvoiceController> logger, IACCService accService)
+        {
+            _logger = logger;
+            _accService = accService;
+        }
+
+        [HttpGet("")]
+        public async Task<IActionResult> Index(int? currentPage, int? pageSize)
```
Comment (timo2o1o) [inline ACCWebAPI/Controllers/IncomingInvoiceController.cs:24]: Please rename this

**Review by timo2o1o — COMMENTED** [review]
I really like your solution, we only need some renaming and we can merge this for the first time.

**Comment by Jusdon** [comment]
@timo2o1o here we are :)

**Comment by timo2o1o** [comment]
> @timo2o1o here we are :)

Nice, thanks! I would rename some of the Properties in Zeitnachweis und Zeitnachweispositionen, but lets do that in a different Issue, to get this monster of a PR done.

**Pull request merged**